File: Lesson_1/TapeEquilibrium/Java/Solution.java
// you can also use imports, for example:
// import java.util.*;

// you can write to stdout for debugging purposes, e.g.
// System.out.println("this is a debug message");

// you can also use imports, for example:
// import java.util.*;

// you can write to stdout for debugging purposes, e.g.
// System.out.println("this is a debug message");

class Solution {
    public int solution(int[] A) {
        // write your code in Java SE 8
        
        int arraySummed = 0;
        int size = A.length;
        int leftSum = 0;
        int minResult = Integer.MAX_VALUE;
        for(int i = 0; i<size; ++i)
        {
            arraySummed += A[i];
            
        }
        
        for(int i=0; i<size-1;++i)
        {
            leftSum+=A[i];
            int result = Math.abs(leftSum - (arraySummed-leftSum));
            if(result<minResult)
            {
                minResult = result;
            }           
        }
        return minResult;
    }
}

File: Lesson_1/TapeEquilibrium/Javascript/solution.js
// you can write to stdout for debugging purposes, e.g.
// console.log('this is a debug message');

function solution(A) {
    // write your code in JavaScript (Node.js 4.0.0)
    var a = A.length;
    var sum = A.reduce(function(pv, cv) { return pv + cv; }, 0);
    var leftSum = 0;
    var minResult = Number.MAX_VALUE;
    for(var i=0, n=A.length-1; i < n; i++) 
    { 
      leftSum += A[i];
      var partialResult = Math.abs(leftSum-(sum-leftSum));
      if(partialResult<minResult){
          minResult = partialResult;
      }
          
    }
    return minResult;
}